infrastructure/main.tf

```diff
@@ -40,7 +42,7 @@ resource "azurerm_resource_group" "rg" {
 }
 
 resource "azurerm_storage_account" "provider" {
-  name                      = "bulkscanning${var.env}"
+  name                      = "${substr(local.account_name, 0, min(length(local.account_name), 24))}"
```

Comment:
This is still equivalent to `${substr(local.account_name, 0, 24))}`

Reply:
I have described the issue in the c9a6c3d8f9570dfa770db850b3273758682296e4's message:
>Terraform's substr() expects that offset + length cannot be larger than the length of the string.

Reply:
I see. I thought it was the reaction to our previous comments on this line:
```
@lgonczar
lgonczar 22 minutes ago  Member
I think it would be best to see an error when the name is too long. Otherwise, we risk having environment component taken out of the name.

  @nitinprabhu
nitinprabhu 20 minutes ago  Member
this will be only for preview I believe so yes it would be good if we can do this only for preview environment and not trim for any other.
```